great_expectations/cli/cli.py

```diff
@@ -87,7 +87,11 @@ def get_command(self, ctx: click.Context, name: str) -> Optional[str]:  # type:
         legacy_module = ""
         if not self.is_v3_api(ctx):
             legacy_module += ".v012"
-
+        if legacy_module == ".v012" and module_name == "upgrade":
+            cli_message(
+                "<red>You are trying to upgrade in the `--v2-api`, but this is no longer supported. If you need to migrate to the latest version, see: https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide/#migrating-to-the-batch-request-v3-api, or use an earlier version of great_expectations<0.15 for old migration support</red>"
```

Comment:
Tiny optional nit suggestion
```suggestion
                "<red>You are trying to upgrade in the `--v2-api`, but this is no longer supported. If you need to migrate to the latest version, see: https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide/#migrating-to-the-batch-request-v3-api, or use an earlier version of great_expectations<0.15 for migration support.</red>"
```